lib/SILOptimizer/Mandatory/TFLowerGraph.cpp

```diff
@@ -1794,12 +1827,32 @@ GLStatus TFGraphLowering::visitGraphOperationInst(GraphOperationInst *inst) {
       // Done with normal attributes.
       break;
     case SILTensorOpInfo::OperandClass::DType: {
-      // This integer value is a dtype.
-      auto val = attrValue.getIntegerValue();
-      TF_SetAttrType(op, name.data(), (TF_DataType)val.getLimitedValue());
+      auto swiftType = attrValue.getMetatypeValue();
+      auto tfType = convertSwiftTypeToTF(swiftType);
+      TF_SetAttrType(op, name.data(), (TF_DataType)tfType);
+      break;
+    }
+    case SILTensorOpInfo::OperandClass::Tensor: {
+      // Tensor can support two cases: an array case (not yet implemented), and
+      // a scalar case.
+      TF_DataType dtype;
+      SmallVector<APInt, 4> elements;
+      SmallVector<int64_t, 4> shape;
+      if (attrValue.getKind() != SymbolicValue::Integer) {
+        llvm_unreachable("FIXME: Tensor-typed attr is not yet handled");
```

Comment:
Please use assert(cond && "message") instead of if+unreachable

Reply:
This code is generalized in the next PR https://github.com/apple/swift/pull/17827/files